containers/datalab/Dockerfile.in

```diff
@@ -22,112 +22,126 @@ RUN echo "deb-src http://ftp.us.debian.org/debian testing main" >> /etc/apt/sour
-        futures==3.0.3 && \
-    pip install -U numpy==1.9.2 && \
-    pip install -U pandas==0.16.2 && \
-    pip install -U scikit-learn==0.16.1 && \
-    pip install -U scipy==0.15.1 && \
-    pip install -U sympy==0.7.6 && \
-    pip install -U statsmodels==0.6.1 && \
-    pip install -U matplotlib==1.4.3 && \
-    pip install -U ggplot==0.6.5 && \
-    pip install -U seaborn==0.6.0 && \
-    pip install -U notebook==4.0.2 && \
-    pip install -U PyYAML==3.11 && \
-    easy_install pip && \
-    find /usr/local/lib/python2.7 -type d -name tests | xargs rm -rf
-
-# Setup Workspace-Sync
-RUN mkdir -p /wsync && \
-    cd /wsync && \
-    wget -nv https://storage.googleapis.com/cloud-datalab/deploy/wsync/20151125/wsync && \
-    chmod u+r,g+x /wsync/* && cd /
+    rm -rf /root/.config/* && \
+    ln -s /root/.config /config && \
+    touch /tools/google-cloud-sdk/lib/third_party/google.py
 
 # Container configuration
 EXPOSE 8080
-VOLUME [ "/var/log/app_engine", "/nb", "/root/.config/gcloud" ]
+VOLUME [ "/notebooks" ]
 
 # Path configuration
 ENV PATH $PATH:/tools/node/bin:/tools/google-cloud-sdk/bin
 ENV PYTHONPATH /env/python
 
-# IPython configuration
-RUN ipython profile create default
-RUN jupyter notebook --generate-config
-
 # Add some unchanging bits - specifically node modules (that need to be kept in sync
 # with packages.json manually, but help save build time, by preincluding them in an
 # earlier layer).
+# Note: ws is now over 1.0 but using that gives issues so leaving at 0.4.2 for now.
 RUN mkdir -p /datalab/web && \
+    mkdir -p /datalab/docs && \
     cd /datalab/web && \
     /tools/node/bin/npm install \
         ws@0.4.32 \
-        http-proxy@1.11.1 \
+        http-proxy@1.13.2 \
         mkdirp@0.5.1 \
-        node-uuid@1.4.3 \
-        bunyan@1.4.0 \
-        socket.io@1.3.6 \
+        node-uuid@1.4.7 \
+        bunyan@1.7.1 \
+        socket.io@1.4.5 \
         tcp-port-used@0.1.2 \
-        node-cache@3.0.0 && \
+        node-cache@3.2.0 && \
     cd / && \
     /tools/node/bin/npm install -g forever
 
-# Install TensorFlow. 
-# Need to install protobuf first due to a TF bug https://github.com/tensorflow/tensorflow/issues/487
-RUN pip install protobuf==3.0.0a3
-RUN gsutil cp gs://cloud-datalab/deploy/tensorflow-0.6.0-cp27-none-linux_x86_64.whl /tmp/tensorflow-0.6.0-cp27-none-linux_x86_64.whl && \
-  pip install /tmp/tensorflow-0.6.0-cp27-none-linux_x86_64.whl && \
-  rm /tmp/tensorflow-0.6.0-cp27-none-linux_x86_64.whl
+# Install ProtoBuf, TensorFlow, DataFlow
+RUN pip install protobuf==3.0.0b2.post2 && \
+    wget https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.7.1-cp27-none-linux_x86_64.whl && \
+    pip install tensorflow-0.7.1-cp27-none-linux_x86_64.whl && \
+    rm tensorflow-0.7.1-cp27-none-linux_x86_64.whl && \
+    easy_install https://github.com/GoogleCloudPlatform/DataflowPythonSDK/archive/v0.2.2.tar.gz && \
```

Comment:
This is an internal version of Dataflow that I would hesitate the send it out, especially with the following tweaks in /usr/local/lib/python2.7/dist-packages/google/__init__.py.